include/swift/AST/ModuleDependencies.h

```diff
@@ -447,55 +460,176 @@ class ModuleDependenciesCache {
 
   /// Retrieve the dependencies map that corresponds to the given dependency
   /// kind.
-  llvm::StringMap<ModuleDependenciesVector> &getDependenciesMap(
-      ModuleDependenciesKind kind);
-  const llvm::StringMap<ModuleDependenciesVector> &getDependenciesMap(
-      ModuleDependenciesKind kind) const;
+  llvm::StringMap<ModuleDependenciesVector> &
+  getDependenciesMap(ModuleDependenciesKind kind);
+  const llvm::StringMap<ModuleDependenciesVector> &
+  getDependenciesMap(ModuleDependenciesKind kind) const;
 
 public:
-  ModuleDependenciesCache() { }
+  GlobalModuleDependenciesCache() {
+    ModuleDependenciesKindMap.insert(
+        {ModuleDependenciesKind::SwiftTextual,
+         llvm::StringMap<ModuleDependenciesVector>()});
+    ModuleDependenciesKindMap.insert(
+        {ModuleDependenciesKind::SwiftBinary,
+         llvm::StringMap<ModuleDependenciesVector>()});
+    ModuleDependenciesKindMap.insert(
+        {ModuleDependenciesKind::SwiftPlaceholder,
+         llvm::StringMap<ModuleDependenciesVector>()});
+    ModuleDependenciesKindMap.insert(
+        {ModuleDependenciesKind::Clang,
+         llvm::StringMap<ModuleDependenciesVector>()});
```

Comment:
I think these inserts could also be replaced by a loop over `FirstKind` to `LastKind`.